client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -683,12 +685,16 @@ define(['./module', 'angular', 'underscore'], function (module, angular, _) {
         budgetLevel = budgetLevel + " to $" + $scope.params.objectives.outcome.budgetrange.maxval;
       }
 
-      $scope.optimizationMessage = optimizationMessageTemplate({
-        checkedPrograms : joinArrayAsSentence(validateObjectivesToMinimize().checkedPrograms, 'name', true),
-        startYear: $scope.params.objectives.year.start,
-        endYear:$scope.params.objectives.year.end,
-        budgetLevel: budgetLevel
-      });
+      if ( budgetLevel && checkedPrograms && startYear && endYear ) {
```

Comment:
This `if` I see al over the place should be testing methods like `$scope.hasProgramBudgetAndYear`  or even  `$scope.hasOptimizationRequisites` 

But is out of scope of the card you took @mkhurramali so don't worry for now. This is something that should be improved on this way too large controller.